assets/js/store/index.ts

```diff
@@ -0,0 +1,272 @@
+import { createReduxStore, register, select, dispatch, createSelector } from '@wordpress/data';
+import { addFilter } from '@wordpress/hooks';
+import * as api from './api';
+import { ContentConnectRelatedEntities, ContentConnectRelationships, ContentConnectState } from './types';
+
+export const STORE_NAME = 'wp-content-connect';
+
+/**
+ * Generates a unique key for related entities based on post ID and relationship key.
+ *
+ * @param postId The ID of the post.
+ * @param relKey The key of the relationship.
+ * @returns A unique key for the related entities.
+ */
+function getRelatedEntitiesKey(postId: number, relKey: string): string {
+	return `related-${postId}-${relKey}`;
+}
+
+/**
+ * Store defaults
+ */
+const DEFAULT_STATE: ContentConnectState = {
+	relationships: {},
+	relatedEntities: {},
+	dirtyEntityIds: new Set(),
+};
+
+type SetRelationshipsAction = {
+	type: 'SET_RELATIONSHIPS';
+	postId: number;
+	relationships: ContentConnectRelationships;
+};
+
+type SetRelatedEntitiesAction = {
+	type: 'SET_RELATED_ENTITIES';
+	key: string;
+	relatedEntities: ContentConnectRelatedEntities;
+};
+
+type MarkPostAsDirtyAction = {
+	type: 'MARK_POST_AS_DIRTY';
+	postId: number;
+};
+
+type ClearDirtyEntitiesAction = {
+	type: 'CLEAR_DIRTY_ENTITIES';
+};
+
+type Action =
+	| SetRelationshipsAction
+	| SetRelatedEntitiesAction
+	| MarkPostAsDirtyAction
+	| ClearDirtyEntitiesAction;
+
+const actions = {
+	/**
+	 * Sets the relationships for a given post ID.
+	 *
+	 * @param postId The ID of the post to set relationships for.
+	 * @param relationships The relationships to set.
+	 * @returns The action to set the relationships.
+	 */
+	setRelationships(postId: number, relationships: ContentConnectRelationships): SetRelationshipsAction {
+		return {
+			type: 'SET_RELATIONSHIPS',
+			postId,
+			relationships,
+		};
+	},
+	/**
+	 * Sets the related entities for a given key.
+	 *
+	 * @param key The key for the related entities.
+	 * @param relatedEntities The related entities to set.
+	 * @returns The action to set the related entities.
+	 */
+	setRelatedEntities(key: string, relatedEntities: ContentConnectRelatedEntities): SetRelatedEntitiesAction {
+		return {
+			type: 'SET_RELATED_ENTITIES',
+			key,
+			relatedEntities,
+		};
+	},
+	/**
+	 * Marks a post as dirty, indicating that it has unsaved changes.
+	 *
+	 * @param postId The ID of the post to mark as dirty.
+	 * @returns The action to mark the post as dirty.
+	 */
+	markPostAsDirty(postId: number): MarkPostAsDirtyAction {
+		// Trigger the block editor to mark the post as dirty.
+		(dispatch('core/editor') as any).editPost({
```

Comment:
Instead of calling dispatch on the string `core/editor` we should import the `store` from the `@wordpress/editor` package. Doing so will allow typescript to understand what's available in the store :) 

```ts
import { store as editorStore } from '@wordpress/editor';

dispatch(editorStore).editPost({});
```
```